### PR TITLE
[CELEBORN-1142] clear shuffleIdCache in shutdown method of ShuffleClientImpl

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### What changes were proposed in this pull request?

As Title

### Why are the changes needed?

As Title

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

PASS GA

### CODE REVIEW

**Comment by jiaoqingbo** [comment]
@ErikFang  @waitinfuture  please cc

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 1]
```diff
@@ -1678,6 +1678,9 @@ public void shutdown() {
     }
     pushExcludedWorkers.clear();
     fetchExcludedWorkers.clear();
+
+    shuffleIdCache.clear();
```
Comment (SteNicholas) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1682]: Move before `pushExcludedWorkers.clear()`?

File: client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java [hunk 2]
```diff
@@ -1676,8 +1676,11 @@ public void shutdown() {
     if (null != lifecycleManagerRef) {
       lifecycleManagerRef = null;
     }
+
+    shuffleIdCache.clear();
     pushExcludedWorkers.clear();
     fetchExcludedWorkers.clear();
+
```
Comment (SteNicholas) [inline client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:1683]: Remove this line change.

**Review by SteNicholas — APPROVED** [review]
LGTM.

**Review by cfmcgrady — APPROVED** [review]

**Comment by cfmcgrady** [comment]
thanks, merging to main(v0.4.0).

**Pull request closed without merging**